lib/execution_engine2/sdk/EE2Runjob.py

```diff
@@ -241,6 +257,100 @@ def _run(self, params, concierge_params):
 
         return job_id
 
+    def _abort_child_jobs(self, child_job_ids):
+        """
+        Cancel a list of child jobs, and their child jobs
+        """
+        for child_job_id in child_job_ids:
+            try:
+                self.sdkmr.cancel_job(
+                    job_id=child_job_id,
+                    terminated_code=TerminatedCode.terminated_by_batch_abort.value,
+                )
+            except Exception as e:
+                # TODO Maybe add a retry here?
+                self.logger.error(f"Couldn't cancel child job {e}")
+
+    def _create_parent_job(self, wsid, meta):
+        """
+        This creates the parent job for all children to mark as their ancestor
+        :param params:
+        :return:
+        """
+        job_input = JobInput()
+        job_input.service_ver = "batch"
+        job_input.app_id = "batch"
+        job_input.method = "batch"
+        job_input.narrative_cell_info = Meta()
+
+        if meta:
+            job_input.narrative_cell_info.run_id = meta.get("run_id")
+            job_input.narrative_cell_info.token_id = meta.get("token_id")
+            job_input.narrative_cell_info.tag = meta.get("tag")
+            job_input.narrative_cell_info.cell_id = meta.get("cell_id")
+            job_input.narrative_cell_info.status = meta.get("status")
+
+        with self.sdkmr.get_mongo_util().mongo_engine_connection():
+            j = Job()
+            j.job_input = job_input
+            j.batch_job = True
+            j.status = Status.queued.value
+            j.wsid = wsid
+            j.user = self.sdkmr.user_id
```

Comment:
Is there no way to initialise these values when you create the job object?

Reply:
@ialarmedalien I don't understand what you mean or why we would want that

Reply:
like
```
j = Job(job_input=job_input, batch_job=True, ...)
```

It doesn't matter though